src/traces/streamtube/calc.js

```diff
@@ -25,11 +25,14 @@ module.exports = function calc(gd, trace) {
     var slen = 0;
     var startx, starty, startz;
     if(trace.starts) {
-        startx = trace.starts.x || [];
-        starty = trace.starts.y || [];
-        startz = trace.starts.z || [];
+        startx = filter(trace.starts.x || []);
```

Comment:
Wouldn't it be better to loop over all the `starts.(x|y|z)` values once and convert all the numeric strings to numbers?

Reply:
Wait, that's what `filter` does essentially.

So we shouldn't need all the `+v` calls downstream of here, right?

Reply:
When the length is known, it converts and copies the raw values.
So during the calculations for length it was needed to cast in few more locations.

Reply:
Ok, so wouldn't it be easier to always convert the raw values? This would be more similar to how we do things in other trace modules e.g. when calling `ax.makeCalcdata` .